### PR TITLE
Create convert_to_grayscale.py

### PR DESCRIPTION
**Pull request by Erfaniaa**
### Describe your change:

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Review by rohan472000 — CHANGES_REQUESTED** [review]
The flake8 error is indicating that the script is using a mix of tabs and spaces for indentation, which is not allowed. It's recommended to use only spaces for indentation, and to configure your text editor or IDE to insert spaces instead of tabs when you press the tab key.

**Comment by rohan472000** [comment]
use below:-

"""
    Implemented an algorithm using opencv to convert a colored image into its grayscale
"""
from cv2 import destroyAllWindows, imread, imshow, waitKey


def convert_to_grayscale(img):
    # getting number of pixels in the image
    pixel_h, pixel_v = img.shape[0], img.shape[1]

    # converting each pixel's color to its grayscale
    for i in range(pixel_h):
        for j in range(pixel_v):
            luminance = (
                0.299 * img[i][j][2] + 0.587 * img[i][j][1] + 0.114 * img[i][j][0]
            )
            img[i][j] = (luminance, luminance, luminance)

    return img


if __name__ == "__main__":
    # read original image
    img = imread("image_data/lena.jpg", 1)

    # convert to its grayscale
    gray = convert_to_grayscale(img)

    # show result image
    imshow("grayscale of original image", gray)
    waitKey(0)
    destroyAllWindows()

**Review by Erfaniaa — COMMENTED** [review]
I converted tabs to spaces

File: digital_image_processing/convert_to_grayscale.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
+"""
+    Implemented an algorithm using opencv to convert a colored image into its grayscale
+"""
+from cv2 import destroyAllWindows, imread, imshow, waitKey
+
+
+def convert_to_grayscale(img):
+    # getting number of pixels in the image
```
Comment (rohan472000) [inline digital_image_processing/convert_to_grayscale.py:8]: add below doctest-

    """
    >>> import cv2
    >>> import numpy as np
    >>> img = np.zeros((100, 100, 3), dtype=np.uint8)
    >>> img[:, :, 0] = 255
    >>> gray = convert_to_grayscale(img)
    >>> assert np.allclose(gray[:, :, 0], 0.114 * 255, rtol=1e-03, atol=1e-03)
    >>> assert np.allclose(gray[:, :, 1], 0.587 * 255, rtol=1e-03, atol=1e-03)
    >>> assert np.allclose(gray[:, :, 2], 0.299 * 255, rtol=1e-03, atol=1e-03)
    """


then add below in main:-
if __name__ == "__main__":
    import doctest
    doctest.testmod()

**Review by rohan472000 — CHANGES_REQUESTED** [review]

File: digital_image_processing/convert_to_grayscale.py (hunk 1 above)
Comment (Erfaniaa) [reply digital_image_processing/convert_to_grayscale.py:8]: What are these?

>>> assert np.allclose(gray[:, :, 0], 0.114 * 255, rtol=1e-03, atol=1e-03)
>>> assert np.allclose(gray[:, :, 1], 0.587 * 255, rtol=1e-03, atol=1e-03)
>>> assert np.allclose(gray[:, :, 2], 0.299 * 255, rtol=1e-03, atol=1e-03)

Are you able to add these things by yourself?

Comment (Erfaniaa) [reply digital_image_processing/convert_to_grayscale.py:8]: Where should exactly "doctest.testmod()" be placed?

Comment (rohan472000) [reply digital_image_processing/convert_to_grayscale.py:8]: under main.

Comment (Erfaniaa) [reply digital_image_processing/convert_to_grayscale.py:8]: In the beginning or in the end?

Comment (Erfaniaa) [reply digital_image_processing/convert_to_grayscale.py:8]: Thank you for your detailed explanation. I added doctest.

**Comment by rohan472000** [comment]
use `ruff .` before every push.

File: digital_image_processing/convert_to_grayscale.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
+"""
+    Implemented an algorithm using opencv to convert a colored image into its grayscale
+"""
+from cv2 import destroyAllWindows, imread, imshow, waitKey
+import doctest
+
+
+def convert_to_grayscale(img):
```
[the remark this answers is not in the document]
Comment (rohan472000) [reply digital_image_processing/convert_to_grayscale.py:8]: give return type in this function.

**Comment by rohan472000** [comment]
still error is causing because of ruff, use `ruff . --fix`

File: digital_image_processing/convert_to_grayscale.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Erfaniaa) [reply digital_image_processing/convert_to_grayscale.py]: Done.

**Comment by Erfaniaa** [comment]
@dynamitechetan Would you please review this pull request?

File: digital_image_processing/convert_to_grayscale.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    Implemented an algorithm using opencv to convert a colored image into its grayscale
+"""
+import doctest
+
+import numpy as np
+from cv2 import destroyAllWindows, imread, imshow, waitKey
+
+
+def convert_to_grayscale(img: np.ndarray) -> np.ndarray:
+    # getting number of pixels in the image
```
Comment (rohan472000) [inline digital_image_processing/convert_to_grayscale.py:11]: add below doctest for `convert-to_grayscale` function

        """ 
        >>> img = np.array([[[255, 0, 0], [0, 255, 0]], [[0, 0, 255], [255, 255, 255]]], dtype=np.uint8)
        >>> convert_to_grayscale(img)
        array([[[ 29,  29,  29],
                [151, 151, 151]],

               [[ 76,  76,  76],
                [255, 255, 255]]], dtype=uint8)

        >>> img = np.array([[[255, 255, 255], [0, 0, 0]], [[128, 128, 128], [192, 192, 192]]], dtype=np.uint8)
        >>> convert_to_grayscale(img)
        array([[[255, 255, 255],
                [  0,   0,   0]],

               [[128, 128, 128],
                [192, 192, 192]]], dtype=uint8)
        """

**Review by rohan472000 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**